### PR TITLE
[WIP] UI rinse

### PR DESCRIPTION
**Pull request by fk**
Still a bit to do, but mostly there 😉 … sneak preview:

image

## TL;DR

1. UI refresh/realign – no major changes to IA, flows, UI logic, markup structure
2. Bump almost all dependencies to latest

### UI things
* switch to a nicer `Cart` icon
* remove `Butler` illustration, replace the `Butler hand` with an arrow icon
* remove the `Sad cart` illustration in favor of just the message
* drop `Futura PT` as `headline` font, switch to `Inter`, add serifs and `monospace` sprinkles
* calmer and friendlier 🤞 UI
  * revamp coupon code badges
  * add a CSS loading indicator/spinner and use it in place of the floating `Butler` head, as well as add it to the `Updating cart` toast
  * show the avatar image for a user logged in via GitHub
  * drop custom `-webkit-scrollbar` styles
  * reduce the color palette
  * drop product descriptions on homepage
  * snappier sidebar transitions
  * refined footer: define a `max-width`, make the links calmer, add `minHeight` constant to increase whitespace (accounted for in the `PageContent` layout via the new constant)  
* align `radius` key nomenclature with other scales and make them friendlier (rounder)
* add `visuallyHidden` style utility (and use it to hide the colons after `Subtotal`, `Shipping` etc. in the cart
* smaller `GatsbyStore` logo (analog to monogram size on gatsbyjs.com; the `Gatsby Store` word mark is controlled individually and still needs some adjustment to match dotcom)

#### component styles
- less magic numbers, more constants (from `styles.js`)
- in `styles.js`
  - reduce number of `colors`, `gradients`
  - add `shadows`, `transitions`, `zIndices`, `fontSizes`, `lineHeights`, `gradients`, `transitions`, and use them
  - add units to the `spacing` scale

#### 404 page
- adjust link to Gatsby homepage from gatsbyjs.org to gatsbyjs.com

#### OpenBar.js
- refactor to get rid of a bunch of magic numbers – `writing-mode` FTW (https://caniuse.com/css-writing-mode)

#### `package.json`
- remove unused `react-onclickoutside`, `react-router-dom`
- bump `gatsby-plugin-image`, `gatsby-source-shopify`
- bump `react`, `react-dom`
- bump `react-icons` to v4
- update `emotion` packages to v11 and remove `babel-plugin-react-icons`
- bump `husky`, `lint-staged`, `prettier`, `stylelint`, `stylelint-config-standard`, `stylelint-order`, `stylelint-processor-styled-components`

#### `gatsby-config.js`
- remove (now) obsolete `apiKey` from the `gatsby-source-shopify` plugin options

#### .eslintrc.json
- remove flowtype plugin and config
- add jsx-a11y plugin

… (WIP)

### CODE REVIEW

File: gatsby-node.js [hunk 1]
```diff
@@ -96,3 +96,9 @@ exports.onCreateWebpackConfig = ({ stage, loaders, actions }) => {
     });
   }
 };
+
+exports.onCreateBabelConfig = ({ actions }) => {
+  actions.setBabelPlugin({
+    name: `babel-plugin-react-icons`
+  });
+};
```
Comment (fk) [inline gatsby-node.js:104]: IIRC, @wardpeet taught me we don't need that stuff anymore – remove.

File: package.json [hunk 2]
```diff
@@ -20,9 +20,11 @@
   "dependencies": {
     "@emotion/core": "^10.0.10",
     "@emotion/styled": "^10.0.11",
+    "@react-icons/all-files": "^4.1.0",
```
Comment (fk) [inline package.json:23]: See comment above (`gatsby-node.js`) – drop `all-files`.

File: package.json [hunk 3]
```diff
@@ -20,9 +20,11 @@
...
     "apollo-boost": "^0.3.1",
     "auth0-js": "^9.10.2",
     "axios": "^0.18.0",
+    "babel-plugin-react-icons": "^0.1.1",
```
Comment (fk) [inline package.json:27]: See comment above (`gatsby-node.js`) – not needed anymore.

Comment (fk) [reply package.json]: Resolved in 82e7b78

File: package.json (hunk 2 above)
Comment (fk) [reply package.json]: Resolved in 82e7b78

File: gatsby-node.js (hunk 1 above)
Comment (fk) [reply gatsby-node.js]: Fixed in 82e7b78

**Comment by fk** [comment]
Closing this – won't get back to this in the near future, and don't want to put anyone through reviewing this, either. Will leave the branch around FWIW. :v:
